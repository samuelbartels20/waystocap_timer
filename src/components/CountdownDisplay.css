.countdown-display {
  font-size: 100px;
  font-weight: bold;
  line-height: 100px;
}

.countdown-display.near-end {
  color: red;
}

.countdown-display.imminent-end {
  animation: blinking 0.3s infinite;
}

@keyframes blinking {
  0%,
  49% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
}
