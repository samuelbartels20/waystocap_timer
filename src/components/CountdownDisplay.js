import React from 'react';
import PropTypes from 'prop-types';

import './CountdownDisplay.css';

function CountdownDisplay({ secondsRemaining, totalSeconds }) {
  const minutes = Math.floor(secondsRemaining / 60)
    .toString()
    .padStart(2, '0');
  const seconds = (secondsRemaining % 60).toString().padStart(2, '0');

  const classNames = ['countdown-display'];

  if (secondsRemaining > 0) {
    if (secondsRemaining < 20) classNames.push('near-end');
    if (secondsRemaining < 10) classNames.push('imminent-end');
  }

  return (
    <div className={classNames.join(' ')}>
      {minutes}:{seconds}
    </div>
  );
}

CountdownDisplay.propTypes = {
  secondsRemaining: PropTypes.number.isRequired,
  totalSeconds: PropTypes.number.isRequired,
};

export default CountdownDisplay;
