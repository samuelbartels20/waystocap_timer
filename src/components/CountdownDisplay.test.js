import React from 'react';
import CountdownDisplay from './CountdownDisplay';
import { shallow } from 'enzyme';

it('renders initial state', () => {
  const initialState = shallow(
    <CountdownDisplay secondsRemaining={0} totalSeconds={0} />
  );
  expect(initialState).toMatchSnapshot();
});

it('renders normal state', () => {
  const initialState = shallow(
    <CountdownDisplay secondsRemaining={25} totalSeconds={30} />
  );
  expect(initialState).toMatchSnapshot();
});

it('renders near end state', () => {
  const initialState = shallow(
    <CountdownDisplay secondsRemaining={15} totalSeconds={30} />
  );
  expect(initialState).toMatchSnapshot();
});

it('renders imminent end state', () => {
  const initialState = shallow(
    <CountdownDisplay secondsRemaining={5} totalSeconds={30} />
  );
  expect(initialState).toMatchSnapshot();
});
