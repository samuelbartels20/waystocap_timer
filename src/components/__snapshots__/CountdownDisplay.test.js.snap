// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders imminent end state 1`] = `ShallowWrapper {}`;

exports[`renders initial state 1`] = `ShallowWrapper {}`;

exports[`renders near end state 1`] = `ShallowWrapper {}`;

exports[`renders normal state 1`] = `ShallowWrapper {}`;
